Added Echis Household Reports and Data sources

andyasne opened the pull request:
The Task is for Changing the dynamic echis reports to a static one for the linked app reports to work properly.
4 Reports
(mobile) Kebele Population Profile Report
(mobile)Household with access to Latrine by type Data Source
(mobile) Household Characteristics
(mobile) Household Profile

andyasne committed:
Added Dynamic Echis Household Reports and Data sources

sravfeyn commented on custom/echis_reports/ucr/data_sources/household_characteristics_data_source.json:
```diff
@@ -0,0 +1 @@
+{"domain": "fmoh-echis-staging", "es_index_settings": {"doc_type": "ElasticSearchIndexSettings", "refresh_interval": "5s", "number_of_shards": 2}, "configured_indicators": [{"comment": null, "create_index": true, "display_name": "health_post_id", "datatype": "string", "expression": {"type": "root_doc", "expression": {"datatype": null, "type": "property_name", "property_name": "health_post_id"}},  ...
```
@andyasne You can use some online json formatter, to generate a readable version of the JSON (with correct spaces and widths). Can you update the two JSON files?

andyasne replied:
I will do that and push again

andyasne committed:
Formatted JSON of the data sources and Reports.

sravfeyn commented on custom/echis_reports/ucr/data_sources/household_characteristics_data_source.json:
```diff
@@ -0,0 +1,501 @@
+{
+   "domain":"fmoh-echis-staging",
+   "es_index_settings":{
```
This is not a valid field, where did you get this from? Can you point me to the UCR source on echisethipia.org?

andyasne replied:
I used the json after clicking the button "Data source source(Advanced)".

andyasne replied:
just sent you the link on slack private message

sravfeyn replied:
@andyasne This comment is still not addressed.

sravfeyn commented on custom/echis_reports/ucr/data_sources/household_characteristics_data_source.json:
```diff
@@ -0,0 +1,501 @@
+      "refresh_interval":"5s",
+      "number_of_shards":2
+   },
+   "configured_indicators":[
```
The config that you find on the web page should be under a field called `config` in this doc. Please see an example [static data source](https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/ucr/data_sources/daily_feeding_forms.json#L20).

andyasne replied:
@sravfeyn  should I wrap the data sources like the data source you sent me .

sravfeyn replied:
This is not addressed.

sravfeyn commented:
There are couple of issues that need to be addressed that I have commented on.

andyasne committed:
json modification echis reports and datasources.  Added JSON elements called domains, server_enviroment and config

andyasne committed:
Cleanup JSON, Removed uneccessary fields

sravfeyn commented:
@andyasne Cool, some of the comments above are still not pending, though? How did you test them.

andyasne commented:
@sravfeyn , I test it by copying the datasource and reports json`s to the local server here that we installed commcare previously.  and It works the same way as the dynamic one. 
which comments did I miss ?

sravfeyn commented:
These comments https://github.com/dimagi/commcare-hq/pull/23256#pullrequestreview-209529238

andyasne commented on custom/echis_reports/ucr/data_sources/household_characteristics_data_source.json:
```diff
@@ -0,0 +1,501 @@
+      "refresh_interval":"5s",
+      "number_of_shards":2
+   },
+   "configured_indicators":[
```
there is a server_environment on that file , line 7

andyasne replied:
@sravfeyn  I think you are reviewing a previous commit ( the first one from the three commits)

andyasne commented on custom/echis_reports/ucr/reports/household_characteristics.json:
```diff
@@ -0,0 +1,286 @@
+      "created_by_builder":true,
+      "builder_source_type":null,
+      "report_builder_version":"2.0",
+      "last_modified":"2019-02-11T14:53:53.646278Z"
```
only found last_modified and removed it

andyasne commented on custom/echis_reports/ucr/reports/household_characteristics.json:
```diff
@@ -0,0 +1,286 @@
+      "doc_type":"ReportMeta",
+      "edited_manually":true,
+      "builder_report_type":"table",
+      "created_by_builder":true,
```
removed it

andyasne commented on custom/echis_reports/ucr/reports/household_characteristics.json:
```diff
@@ -0,0 +1,286 @@
+   ],
+   "custom_query_provider":null,
+   "soft_rollout":"0",
+   "_id":"19c109ee62babf4c2b7ee7fdd7cca4a0",
```
removed it

andyasne commented on custom/echis_reports/ucr/data_sources/household_profile_data_source.json:
```diff
@@ -505,18 +515,18 @@
    "doc_type":"DataSourceConfiguration",
    "icds_rebuild_related_docs":false,
-   "table_id":"household_profile_data_source",
+   "table_id":"static-household_profile_data_source",
```
did that for consistency with other already made data source

andyasne committed:
Revert "resolved Merged Conflicts"

This reverts commit aacc4776b162a22f25d69d195c0f023f843975a2.

andyasne committed:
resolved Merged Conflicts

andyasne committed:
Revert "resolved Merged Conflicts"

This reverts commit aacc4776b162a22f25d69d195c0f023f843975a2.

andyasne commented:
@sravfeyn I fixed it as written. Pushed too.

sravfeyn commented on custom/echis_reports/ucr/data_sources/household_characteristics_data_source.json:
```diff
@@ -0,0 +1,510 @@
+   "named_filters":{
+
+   },
+   "_id":"02f1aecf16cae01da886e3fcea186f66",
```
This should be removed.

sravfeyn committed:
Remove extra fields

sravfeyn commented:
@andyasne Thanks for updating.

There are still some extra fields that are not required. I updated one of the UCRs and made a commit here to remove such fields. Here is the [commit](https://github.com/dimagi/commcare-hq/pull/23256/commits/baa4a98aa4defc09c350017ecec80db2f49846af). Can you update all other UCRs to look like this?

andyasne commented:
yes I will update the UCR based on the commit and let you know. Thank you

andyasne committed:
Removed es_index_settings and Empty Named Filters

andyasne committed:
Resolved Merge Conflict

andyasne commented:
@sravfeyn  I edited the files as suggested and pushed.
[Here is the Commit](https://github.com/dimagi/commcare-hq/pull/23256/commits/722da77eb4b5e7847a5a525ae483e6bd37977f46)

andyasne committed:
Removed JSON elements :meta,sql_settings,engine_id,backend_id,disable_destructive_rebuild,doc_type,icds_rebuild_related_docs

andyasne committed:
Fix report hmis monthly Ind Fp- Changed report fp-10.. to fp-30, Tag: custom

andyasne committed:
Merge branch 'master' of https://github.com/andyasne/commcare-hq into Fix-REPORT-HMISMonthlyInd

andyasne committed:
changed report column health post id visibility to false

changed report column health post id visibility to false

andyasne committed:
changed report column health post id visibility to false

andyasne committed:
changed report column health post id visibility to false

andyasne committed:
changed report column health post id visibility to false

andyasne committed:
changed report column health post id visibility to false

sravfeyn commented on custom/icds_reports/tasks.py:
```diff
@@ -256,6 +257,7 @@ def move_ucr_data_into_aggregation_tables(date=None, intervals=2):
                             ).apply_async()
 
             res_awc.get()
+            _bust_awc_cache.delay()
```
Why are you changing this, this is not related to eCHIS, please remove this.

andyasne replied:
I don`t know how that change is made... I didn`t touch any other files other than the 8 ones that we are working on...checked all my commits and I was not able to find that change.

andyasne replied:
I think that line of code and others on that file are added after I opened this pull request.  Check this commit
Revision: ac177a07ee1e6a6410996adf0d6f179b7e3afbbd
Author: Cal Ellowitz <calellowitz@gmail.com>
Date: 2/13/2019 5:13:44 PM
Message:
pull cache bust out of main task

andyasne commented on custom/echis_reports/ucr/data_sources/household_profile_data_source.json:
```diff
@@ -0,0 +1,527 @@
+
+   },
+   "is_deactivated":false,
+   "sql_settings":{
```
??  which commit is this ?  "is_deactivated" or "is_deactivated" does not exist on that file and also on other data sources too.

andyasne commented on custom/echis_reports/ucr/data_sources/kebele_population_profile_data_source.json:
```diff
@@ -0,0 +1,1237 @@
+   "named_expressions":{
+   },
+ 
+   "sql_column_indexes":[
```
I did notice this but I saw that this empty json elements are added on the static reports that you guys did it previously. fearing that it might crush the app I left it there. I will remove that and push now.

andyasne committed:
removed empty named_expressions and sql_column_indexes from all datasources

andyasne committed:
Resolved Merge Conflict

Resolved Merge Conflict

andyasne committed:
changed fp_at_10_to_14 to fp_at_30_to_49

changed fp_at_10_to_14 to fp_at_30_to_49

sravfeyn commented:
Ok, please merge the master and let the build pass.

andyasne committed:
Fix for stickler-ci — Lint errors found

error msg "W391 blank line at end of file"
Fix: removed blank line at the end of the file

andyasne commented:
Fix for Label requirements: REQUIRED_LABELS_ANY

andyasne committed:
Fix for Label requirements : REQUIRED_LABELS_ANY: any, any_2

Fix for Label requirements 
REQUIRED_LABELS_ANY: any, any_2

andyasne commented:
@sravfeyn , the changes are already on my master branch. I don`t have a privilege to work on ( merge) to dimagi:master branch.
Is that what you  meant by "merge the master"?

sravfeyn commented:
I mean that you should merge `dimagi:master` into `andyasne:master`. Please see this [documentation](https://help.github.com/en/articles/merging-an-upstream-repository-into-your-fork)

andyasne committed:
Resolved Merge Conflicts

andyasne committed:
Resolved Merge conflicts

andyasne committed:
Resolved Merge conflicts

andyasne committed:
Resolved merge conflicts

andyasne commented:
@sravfeyn ,I merged dimagi master to andyasne master and pushed.

sravfeyn commented:
Great, thanks. Please ;et the build pass and then we can merge it.

sravfeyn commented:
@andyasne FYI, the build didn't pass for some reason, I have restarted it.

andyasne commented:
ya @sravfeyn , I saw that,  I am checking out the Travis log to see if my changes somehow brake the build.

sravfeyn commented:
The build failed again, I have kicked it again. If it fails again, can you please merge with latest dimagi master again?

andyasne commented:
yes, I will merge it again.

andyasne committed:
Resolved Merge Conflict

andyasne committed:
Resolved Merge Confilct

andyasne committed:
merged with dimagi`s master

andyasne committed:
merged with the dimagi`s master

andyasne committed:
removed changes made to /icds_reports/tasks.py while merging with master

sravfeyn commented:
It looks like it has conflicts again.

sravfeyn commented:
@andyasne I still see you are having issues getting it to pass, please let me know if I can help with anything.

andyasne committed:
Merge https://github.com/dimagi/commcare-hq

andyasne commented:
ya @sravfeyn  , I have been trying to merge and push...but the build always fails. I don`t know if either the main branch is failed or because of the changes added by the pull request.

sravfeyn commented:
I think for some reason, these still don't reflect the latest state of the dimagi repository. There are changes to [this file](https://github.com/dimagi/commcare-hq/pull/23256/files#diff-b8c5a5e308a203bcaee8ed83e6837edf) which shouldn't be there if it's properly merged. Can you try this as well https://help.github.com/en/articles/syncing-a-fork

andyasne committed:
Merge remote-tracking branch 'upstream/master'

andyasne committed:
Merge remote-tracking branch 'upstream/master'

andyasne committed:
Merge remote-tracking branch 'upstream/master'

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

andyasne committed:
Merge remote-tracking branch 'upstream/master'

andyasne commented:
@sravfeyn I tried many times merging as the provided link and  pushing  , but the build still fails .
 don`t know what to do with this.

andyasne commented:
@sravfeyn I am going to create another new fork from dimagi`s latest commcare, create another new pull request, check if the build is  successful before I add the new files(from here), then create a new commit and check if the build is successful.
